src/computer_player.c

```diff
@@ -0,0 +1,604 @@
+    struct DECK* last = *head;
+
+    newNode->card = card;
+    newNode->next = NULL;
+    if (*head == NULL) {
+        *head = newNode;
+        return;
+    }
+    while (last->next != NULL)
+        last = last->next;
+    last->next = newNode;
+    return;
+}
+
+
+/** @brief Pick the best card to discard from hand cards based on
+ *         the newest card on the table.
+ *         if it is an action card:
+ *              - if it is a draw/draw2 card and number of the card of the color of that draw/draw2 card < 2,
+ *                use wild card if possible to change to color
+ *              - if it is a wild card, play the corresponding color
+ *              - if do not have the corresponding color, draw card 
+ *         if opponent player has more than one card # > 1:
+ *          case 1 : no matched color and no matched number
+ *              - if have more than one wild card, play wild card with color with the most occurence
+ *              - if have skip, use skip card 
+ *              - if don't have wild or skip, draw card from pile
+ *          case 2 : have the matched color and no matched number
+ *              - choose the largest number on that color to make opponent lose more point
+ *              - if matched color > 2, use match draw/draw2/skip card based on the color with 
+ *                the most occurence
+ *          case 3 : no matched color, but have matched number
+ *              - if have multiple matched number, choose the color with the most occurence 
+ *                or corresponding action card
+ *              - if have only one matched number with different card, get rid of it
+ *          case 4 : have both matched color and match number
+ *              - Quickly get rid of matched numbers in the hand.
+ *              
+ *          if opponent player has one card left: 
+ *              - use action card if possibile
+ *              - avoid play the last color that the player played
+ *
+ *  @return A best cooresponding card from hand cards, or call an action (draw card)
+ */
+
+void pick_action(struct CARD inputCard, struct DECK *hand_card)
+{
+    int wildCount= 0, colorCount=0, numberCount=0;
+    char most_color;
+    char choosed_color;
+    struct CARD playCard, DrawCard;
+    struct DECK *temp = hand_card;
+    int caseNum = pick_case(inputCard, hand_card);
+    most_color = find_most_color(hand_card);
+    switch (caseNum)
+    {
+    case 1:
+        /* no matched color or number, check if wild is > */
+        while (temp->next != NULL)
+        {
+            if (temp->card.name == WILD)
+            {
+                wildCount++;
+            }
+            temp = temp->next;
+        }
+        /* if wild card >1, play the WILD with color with the most occurence */
+        if (wildCount > 1)
+        {
+            choosed_color = most_color;
+            playCard.name = WILD;
+            playCard.color = choosed_color;
```

Comment:
choosed_color = most_color;  -- delete it.
playCard.color = choosed_color;  change to playCard.color = most_color